test/trento_web/views/v1/sap_system_view_test.exs

```diff
@@ -2,41 +2,130 @@ defmodule TrentoWeb.V1.SapSystemViewTest do
+        )
+
+      expected_sap_system_json =
+        [
+          %{
+            id: sap_system.id,
+            tags: [],
+            inserted_at: nil,
+            updated_at: nil,
+            health: sap_system.health,
+            tenant: sap_system.tenant,
+            sid: sap_system.sid,
             database_instances: [
-              %DatabaseInstanceReadModel{
-                system_replication: "Secondary",
-                system_replication_status: "",
-                tenant: "ed758b1b-358b-47cf-ae99-f9e17da0cc54"
-              },
-              %DatabaseInstanceReadModel{
-                system_replication: "Primary",
-                system_replication_status: "ACTIVE",
-                tenant: "5d4f71dd-b672-470c-b9de-79d2b943a9fd"
+              %{
+                sap_system_id: database_id,
+                # keep backward compatibility between sap_system_id and database_id
+                database_id: database_id,
+                sid: database_instance.sid,
+                tenant: database_instance.tenant,
+                instance_number: database_instance.instance_number,
+                instance_hostname: nil,
+                features: database_instance.features,
+                http_port: nil,
+                https_port: nil,
+                start_priority: nil,
+                host_id: database_instance.host_id,
+                system_replication: database_instance.system_replication,
+                system_replication_status: database_instance.system_replication_status,
+                health: database_instance.health,
+                absent_at: nil,
+                inserted_at: nil,
+                updated_at: nil
               }
             ],
-            application_instances: []
+            database_id: database_id,
+            application_instances: [
+              %{
+                sap_system_id: application_instance.sap_system_id,
+                sid: application_instance.sid,
+                instance_number: application_instance.instance_number,
+                features: application_instance.features,
+                host_id: application_instance.host_id,
+                health: application_instance.health,
+                absent_at: nil,
+                inserted_at: nil,
+                updated_at: nil,
+                start_priority: nil,
+                instance_hostname: nil,
+                https_port: nil,
+                http_port: nil
+              }
+            ],
+            ensa_version: sap_system.ensa_version,
+            database_sid: database_sid,
+            db_host: sap_system.db_host
           }
         ]
-      })
-
-    assert Enum.any?(database_instances, fn
-             %{
-               system_replication: "Secondary",
-               system_replication_status: "ACTIVE"
-             } ->
-               true
-
-             _ ->
-               false
-           end)
+
+      assert ^expected_sap_system_json =
```

Comment:
Some comments about this. If you do a simple pattern matching `=` test, you don't really need to have all the fields of this `expected_sap_system_json`.
If you want to do a real "everything is equal", you should use the `==` comparison.

Besides that, I encourage you to now have a `expect_**` variable. It hides things.
Put just the expected values there, and you see all of the pinned things.
But with `==` you don't need to pin anything anyway